modules/apigee-x-instance/variables.tf

```diff
@@ -34,12 +34,12 @@ variable "apigee_org_id" {
   type        = string
 }
 
-variable "cidr_mask" {
-  description = "CIDR mask for the Apigee instance."
-  type        = number
+variable "ip_range" {
+  description = "Customer-provided CIDR block of length 22 for the Apigee instance."
+  type        = string
   validation {
-    condition     = contains([16, 20, 22], var.cidr_mask)
-    error_message = "Invalid CIDR mask; Allowed values for cidr_mask: [16, 20, 22]."
+    condition     = cidrnetmask(var.ip_range) == "255.255.252.0"
```

Comment:
nit: is it worth checking for a `null` value, or specifying the new `nullable` modifier?

Reply:
@ludoo We don't need to check for `null` values as all the possible scenarios are handled by `cidrnetmask()` function.
If the `ip_range` variable is `null` or does not match the pattern `X.X.X.X/22` the validation will fail.

Reply:
What I meant is guarding the condition so it does throw an error like this

```
> cidrnetmask(null)
╷
│ Error: Invalid function argument
│ 
│   on <console-input> line 1:
│   (source code not available)
│ 
│ Invalid value for "prefix" parameter: argument must not be null.
```

Reply:
`condition = var.ip_range != null && cidrnetmask(var.ip_range) == "255.255.252.0"`

Reply:
Whether or not we add `var.ip_range != null` in the `condition` block, the error message remains the same while doing a `terraform plan` which is as below:

**Case# 1**: `var.ip_range != null` check added
```
$ terraform plan
╷
│ Error: Invalid function argument
│ 
│   on variables.tf line 41, in variable "ip_range":
│   41:     condition     = var.ip_range != null && cidrnetmask(var.ip_range) == "255.255.252.0"
│     ├────────────────
│     │ var.ip_range is null
│ 
│ Invalid value for "prefix" parameter: argument must not be null.
```

**Case# 2**: `null` check not added
```
$ terraform plan
╷
│ Error: Invalid function argument
│ 
│   on variables.tf line 41, in variable "ip_range":
│   41:     condition     = cidrnetmask(var.ip_range) == "255.255.252.0"
│     ├────────────────
│     │ var.ip_range is null
│ 
│ Invalid value for "prefix" parameter: argument must not be null.
```

Reply:
@ludoo I have modified the condition as below:
`try(cidrnetmask(var.ip_range), null) == "255.255.252.0"`

This way the error thrown by `cidrnetmask` function is being handled and we get the custom error message when validation fails.

```
$ terraform plan
╷
│ Error: Invalid value for variable
│ 
│   on variables.tf line 37:
│   37: variable "ip_range" {
│ 
│ Invalid CIDR block provided; Allowed pattern for ip_range: X.X.X.X/22.
│ 
│ This was checked by the validation rule at variables.tf:40,3-13.
```

Reply:
gah, I forgot about Terraform following both branches even if the first part of the condition is false :facepalm: 

Let me reiterate this is just nitpicking, so feel free to merge at any time :)

I would just set `nullable = false` in the variable, bump the minimum Terraform version in `versions.tf` to `>= 1.1.0`, and call it a day. :)